-1

Tested at: d99fc8244a8b9772788f85ddf5eabf2ba9c68acd

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3e42fd/1091/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation error when building: 

<pre>>> Entering Package Validation/CaloTowers
Entering library rule at Validation/CaloTowers
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-06-20-2300/src/Validation/CaloTowers/src/CaloTowersClient.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-06-20-2300/src/Validation/CaloTowers/src/CaloTowersValidation.cc 
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-06-20-2300/src/Validation/CaloTowers/src/CaloTowersValidation.cc: In constructor 'CaloTowersValidation::CaloTowersValidation(const edm::ParameterSet&)':
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-06-20-2300/src/Validation/CaloTowers/src/CaloTowersValidation.cc:34:7: error: no match for 'operator!' (operand type is 'std::__cxx11::string {aka std::__cxx11::basic_string<char>}')
   if (!mc_)
       ^~~~
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-06-20-2300/src/Validation/CaloTowers/src/CaloTowersValidation.cc:34:7: note: candidate: operator!(bool) <built-in>
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-06-20-2300/src/Validation/CaloTowers/src/CaloTowersValidation.cc:34:7: note:   no known conversion for argument 1 from 'std::__cxx11::string {aka std::__cxx11::basic_string<char>}' to 'bool'
gmake: *** [tmp/slc7_amd64_gcc700/src/Validation/CaloTowers/src/ValidationCaloTowers/CaloTowersValidation.cc.o] Error 1
</pre>

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>++ scram build outputlog
Reading cached build data
Entering library rule at Validation/CaloTowers
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-06-20-2300/src/Validation/CaloTowers/src/CaloTowersClient.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-06-20-2300/src/Validation/CaloTowers/src/CaloTowersValidation.cc 
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-06-20-2300/src/Validation/CaloTowers/src/CaloTowersValidation.cc:34:7: error: invalid argument type 'std::string' (aka 'basic_string<char>') to unary expression
  if (!mc_)
      ^~~~
1 error generated.
gmake: *** [tmp/slc7_amd64_gcc700/src/Validation/CaloTowers/src/ValidationCaloTowers/CaloTowersValidation.cc.o] Error 1
++ /build/cmsbld/jenkins/workspace/ib-run-pr-tests/cms-bot/buildLogAnalyzer.py --logDir /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-06-20-2300/tmp/slc7_amd64_gcc700/cache/log/src
</pre>

--- a/Validation/CaloTowers/interface/CaloTowersValidation.h
+++ b/Validation/CaloTowers/interface/CaloTowersValidation.h
@@ -40,7 +40,7 @@ private:
 
   std::string outputFile_;
   std::string hcalselector_;
-  std::string mc_;
+  bool mc_;
   bool useAllHistos_;
 
   typedef math::RhoEtaPhiVector Vector;

